src/main/java/org/broadinstitute/hellbender/tools/walkers/filters/VariantFiltration.java

```diff
@@ -323,35 +329,27 @@ private void filter(final VariantContext vc, final FeatureContext featureContext
         writer.add(builder.make());
     }
 
-    private GenotypesContext makeGenotypes(final VariantContext vc) {
-        final GenotypesContext genotypes = GenotypesContext.create(vc.getGenotypes().size());
-
-        // for each genotype, check filters then create a new object
-        for ( final Genotype g : vc.getGenotypes() ) {
-            if ( g.isCalled() ) {
-                final List<String> filters = new ArrayList<>();
-                if ( g.isFiltered() ) {
-                    filters.add(g.getFilters());
-                }
-
-                // Add if expression filters the variant context
-                for ( final JexlVCMatchExp exp : genotypeFilterExps ) {
-                    if ( invertLogic(VariantContextUtils.match(vc, g, exp), invertGenotypeFilterExpression) ) {
-                        filters.add(exp.name);
-                    }
-                }
+    /**
+     * Get the genotype filters
+     *
+     * @param vc the variant context
+     * @param g the genotype
+     * @return list of genotype filter names
+     */
+    private List<String> getGenotypeFilters(final VariantContext vc, final Genotype g) {
+        final List<String> filters = new ArrayList<>();
+        if (g.isFiltered()) {
+            filters.add(g.getFilters());
```

Comment:
`getFilters` returns a single string which is going to cause subtle sorting issues when combined with the jexl filters.  GenotypeBuilder.filters() sorts the list of strings if there are multiple strings, but it will have a single block of existing filters and then separate jexl ones, so it won't end up sorting correctly.

Could you split the filters so that they will end up sorting correctly?  You could run the jexl ones first and then only split if any of those were added for efficiency if you think it's necessary.